### ISSUE TITLE
Driver signing

### ISSUE THREAD

**Issue by ghost** [BenjaminKim/dokanx#15]
Eventually signed drivers are going to be needed:

Here's a potential workaround for a while: https://github.com/koush/UniversalAdbDriver/blob/master/UniveralAdbDriverInstaller/Program.cs

Basically, create a certificate that's distributed with an installer. Before the driver is installed, the certificate is added to the root store.

**Comment by Shamar** [comment]
ReactOS sign opensource drivers for free. 

You should contact them: https://www.reactos.org/wiki/Driver_Signing

**Comment by BenjaminKim** [comment]
@Shamar I didn't know about that. Thank you for informing this. I will try that.

**Comment by Shamar** [comment]
Please, let us know about any progress about this matter. Driver signature is the only issue that is preventing our team to adopt dokanx for a few internal tools.

**Comment by lalanikarim** [comment]
My employers are evaluating DokanX as a possible platform to help create a Windows mountable Virtual File System (Fuse and OSXFuse has served us well for Linux and Mac OS X). Distribution will be an issue unless the Kernel Driver is signed. Please keep us updated on any progress on this front.

**Comment by Liryna** [comment]
https://github.com/Maxhy/dokany have a signed version of Dokan synced with DokanX.

**Comment by Shamar** [comment]
@jimmy00784 are you planning to use/join/start an open source project? 
I'm designing exactly the same thing.

**Comment by lalanikarim** [comment]
@Shamar Not exactly. It is for use within a non-open source commercial project. There are other open source projects we are utilizing without altering their sources as library dependencies and including the necessary Licenses to go with them.

**Comment by BenjaminKim** [comment]
I have sent an email to ReactOS foundation. I am not sure they accept this yet. I will keep informing that.

**Comment by lalanikarim** [comment]
Any update on this yet?

**Comment by BenjaminKim** [comment]
@jimmy00784 Unfortunately, not yet.

**Comment by jimmy00784** [comment]
@BenjaminKim We are only weeks from launching our product to market and a signed driver is essential to providing virtual FS support under Windows. Our only other option would be to rewrite our own FS driver outside of DokanX. 

My employers have offered to assist with getting the DokanX driver signed. Let me know if you are interested pursue this path at your earliest.

**Comment by suy** [comment]
El Friday 27 March 2015, Karim Lalani escribió:

> @BenjaminKim We are only weeks from launching our product to market and a
> signed driver is essential to providing virtual FS support under Windows.
> Our only other option would be to rewrite our own FS driver outside of
> DokanX.
> 
> My employers have offered to assist with getting the DokanX driver signed.
> Let me know if you are interested pursue this path at your earliest.

Have you given a try to DokanY's driver? The project, driver wise, seems 
aligned completely. The difference is in the library: DokanX has support for 
std::function instead of a plain C callback. If you can resort to the latter, 
or use a wrapper, I _think_ you can achieve at least source and behaviour 
compatibility.

**Comment by fearthecowboy** [comment]
I can sign drivers thru the Outercurve Foundation.

Let me spin up my Win7 development VM and see if I can build this. If I get the build running right, it's trivial for me to sign it.

G

**Comment by ghost** [comment]
Good to see you here, Garrett. Is the signing something that could be put
into a CI system?

**Comment by fearthecowboy** [comment]
@voltagex Um, I could probably put a CI system in place for it, I should do that for OneGet anyway.

**Comment by ghost** [comment]
Any updates on this @fearthecowboy? (and a couple of other things I asked you a while back...)

**Comment by SoulMagistic** [comment]
@voltagex You should look at DokanY (https://github.com/dokan-dev/dokany)
The project is probably the most serious fork of dokan and kindly signed by a company.
It also looks like the maintainers are using Dokan their self, so it give a good vision of the stability and evolution of the DokanY fork.
You can also see that they have contact all project arround Dokan to recreate the community.

I was a DokanX user and I moved to DokanY in a minute since there is no big change.

**Comment by marinkobabic** [comment]
I will sign the driver and make a pull request next week. So you will have signed drivers for every release

**Comment by marinkobabic** [comment]
So please <a href="http://liveupdate.ruf.ch/transfer/dokanx_driver.zip" target="_blank">download this driver</a> for dokanx
